platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -72,10 +72,22 @@ public ConfigurationFilesPoller(@NotNull Path configurationDirectoryPath,
         this.fileExtension = fileExtension;
     }
 
-    public void init() {
+    public void init() throws IOException {
         LOGGER.debug("Starting {}...",
                 this.getClass()
                         .getName());
+
+        try {
+            LOGGER.debug("Registering path [{}] with Watch Service.",
+                    configurationDirectoryPath.toString());
+            configurationDirectoryPath.register(watchService, ENTRY_CREATE);
+        } catch (IOException e) {
+            LOGGER.error("Unable to register path [{}] with Watch Service",
+                    configurationDirectoryPath.toString(),
+                    e);
+            throw e;
```

Comment:
If the `ConfigurationFilesPoller` fails to properly initialize, we could end up with an application that's misconfigured or missing managed services. I felt that simply logging an error if we couldn't monitor the directory wasn't enough and that actually failing the container was a better option here (best effort vs. fail fast). But I'm open to other points of view :-)